tests/sweep_framework/runner.py

```diff
@@ -35,8 +35,22 @@ def get_username():
     return os.environ["USER"]
 
 
+def get_devices(test_module):
+    try:
+        device = test_module.device_fixture()
+        return device, True
+    except:
+        device = ttnn.open_device(0)
+        return device, False
+
+
 def run(test_module, input_queue, output_queue):
-    device = ttnn.open_device(0)
+    device_generator, custom_device = get_devices(test_module)
```

Comment:
Instead of custom device boolean, maybe it would be useful to modify the spec. to return a name. For example t3k, tgg, wh_n300 etc.

In that case we wouldn't have "custom" devices... all would be named.


@jdesousa-TT what do you think?

Reply:
Then for tear down, we always use a device fixture... if none is specified, we just return a default one that is defined within a util-like module... `device_fixtures.py`?

Reply:
I like the default fixture idea, no device open in the runner at all. I’ll define a default fixture then just sub in the custom fixture if it exists. Simplifies the runner, moves the logic out. Thx.